### ISSUE TITLE
Logical margin properties on .alignleft and .alignright

### ISSUE THREAD

**Issue by Azragh** [WordPress/gutenberg#40501]
### Description

Just fount this in the editor stylesheet:
```
.editor-styles-wrapper .edit-post-visual-editor__post-title-wrapper > .alignleft, 
.editor-styles-wrapper .block-editor-block-list__layout.is-root-container > .alignleft {
    float: left;
    margin-inline-start: 0;
    margin-inline-end: 2em;
}
```

Edit: So in my opinion, the classes `.alignleft` and `.alignright` should also have `margin-left` and `margin-right` properties (none that depend on the reading direction), because even on a Chinese website, for example, left remains left. If the functionality of `.alignleft` and `.alignright` were completely adapted to the reading direction, one could use these properties. However, this would mean that the float properties would also have to be changed to `inline-start` and `inline-end`. This could work, but then the the toolbar icons and the naming itself would not make sense anymore / had to be adjusted for other languages somehow (at least the icon should represent the new float direction then).

### Step-by-step reproduction instructions

Add a block, change alignment to left or right and inspect it.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by ndiego** [comment]
Hi @Azragh are you able to elaborate on this issue a bit more?

**Comment by Azragh** [comment]
@ndiego I can try. So in my opinion, the classes .alignleft and alignright should also have margin-left and margin-right properties (none that depend on the reading direction), because even on a Chinese website, for example, left remains left. If the functionality of .alignleft and .alignright were completely adapted to the reading direction, one could use these properties. However, this would mean that the float properties would also have to be changed to "inline-start" and "inline-end". Not a big problem, but then the the toolbar icons and the naming itself don't make sense / have to be adjusted for other languages..

Since these classes are around for ages and are used by many (classic) themes, I would advise against making these changes.

**Comment by Azragh** [comment]
@ndiego Moved my comment to the description.

**Comment by tellthemachines** [comment]
Agreed, it makes sense to use equivalent values for float and its corresponding margin. 

Note that CSS logical properties should soon be usable with float too, but currently [browser support is not widespread](https://developer.mozilla.org/en-US/docs/Web/CSS/float#browser_compatibility). Whether that would be a better route (using logical properties for _both_ float and margin) is another matter, and it would require rethinking the icons on the controls too. 

For now, the best approach is to update the margins to use left/right.

**Comment by Azragh** [comment]
> For now, the best approach is to update the margins to use left/right.

That's exactly what I was trying to say. I adjusted the property in my themes for specificity, but it just didn't feel quite right because left and right are directional and have nothing to do with the reading direction.

When the browser support is ready, it would be best to introduce new classes for floats depending on the reading direction, strongly agree. Regarding icons, I could well imagine that the existing icon can also be used for other languages or reading directions - it would only have to be rotated / mirrored according to the set language. 

The only question is whether the existing classes `.alignleft`, `.alignright` and `.aligncenter` would then no longer be available / would be replaced with the new ones. A solution for this would be a setting in the theme.json, in which you can choose whether alignments should be dependent on the reading direction. Off by default, i.e. the usual classes `.alignleft` etc, but if switched on, the icons adapt and new classes such as `.align-inline-start` are used.

Something like this, maybe with small adjustments:

![image](https://user-images.githubusercontent.com/15014142/236754246-1316f41c-c857-4da5-af4d-da864f6c3f36.png)

Or a little less modern, kinda like in MS Word (really quick sketch):

![image](https://user-images.githubusercontent.com/15014142/236758771-3cd581f8-0790-4f21-bdbc-6cade91bbf1c.png)

**Comment by jasmussen** [comment]
Coming back to this one, it sounds like the action item is to update the margins to use left/right, as opposed to logical margin-inline-*, (so in right-to-left languages the gap lands on the wrong side). Is that right? CC: @tellthemachines in case you have instincts. If yes, for now I've added a label.

**Comment by Azragh** [comment]
@jasmussen yes, atm the naming of ".alignleft" and ".alignright" just doesn't make sense if it uses logical properties. And changing the class name would break backwards compatibility so imo it has to remain left and right.
